plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -75,7 +75,7 @@ interface TypeEmbedding {
     fun accessInvariants(): List<TypeInvariantEmbedding> = emptyList()
 
     // Note: this function will replace accessInvariants when nested unfold will be implemented
-    fun predicateAccessInvariants(): List<TypeInvariantEmbedding> = emptyList()
+    fun predicateAccessInvariant(): TypeInvariantEmbedding? = null
```

Comment:
Is a type invariant always exactly one predicate?  What about read-only vs write; will we have different functions for getting those?

Reply:
yes, the plan is to have two different functions, one for read access and one for write access